components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -60,16 +59,16 @@ fn check_second(second: u32) -> Result<u32> {
 }
 
 #[inline]
-fn check_micros(micros: u32) -> Result<u32> {
-    if micros > MAX_MICROS {
-        Err(Error::truncated_wrong_val("MICROS", micros))
+fn check_nanos(nanos: u32) -> Result<u32> {
+    if nanos > MAX_NANOS {
+        Err(Error::truncated_wrong_val("NANOS", nanos))
     } else {
-        Ok(micros)
+        Ok(nanos)
     }
 }
 
 mod parser {
-    use super::{check_hour, check_minute, check_second, Error, MICRO_WIDTH, TEN_POW};
+    use super::{check_hour, check_minute, check_second, Error, NANO_WIDTH, TEN_POW};
```

Comment:
Maybe we could upgrade the `nom` to the latest version in the parser.

Reply:
Maybe later. Otherwise we have to also refactor the time module.